src/core/or/circuituse.c

```diff
@@ -1680,6 +1681,7 @@ circuit_testing_failed(origin_circuit_t *circ, int at_last_hop)
 void
 circuit_has_opened(origin_circuit_t *circ)
 {
+  tor_trace(circuit, opened, circ);
```

Comment:
Is this typechecked only when building with lttng?

Reply:
Ah yes good observation. LTTng does type checking but afaik, not the USDT probes (Dtrace, Systemtap, etc...). These tracers will eat what is given and then you need to define the parameter type outside with some other file specific to DTrace.

So right now, that event can be a USDT probes (if enabled) but the object passed will be considered a simple pointer, nothing gets deref or anything. We need to configure that per tracepoint. I haven't done that job for other tracers since supporting them all is quite some work...